src/jit/gentree.h

```diff
@@ -4626,24 +4636,58 @@ struct GenTreeAllocObj final : public GenTreeUnOp
 // be defined already.
 //------------------------------------------------------------------------
 
-inline bool GenTree::OperIsBlkOp() const
+inline bool GenTree::OperIsBlkOp()
 {
-    return (gtOper == GT_INITBLK || gtOper == GT_COPYBLK || gtOper == GT_COPYOBJ);
+    return (((gtOper == GT_ASG) && varTypeIsStruct(gtOp.gtOp1))
+#ifndef LEGACY_BACKEND
+            || (OperIsBlk() && (AsBlk()->Data() != nullptr))
```

Comment:
Will do.

---
In reply to: [77389887](https://github.com/dotnet/coreclr/pull/7027#discussion_r77389887) [](ancestors = 77389887)